test/unit/index.ts

```diff
@@ -137,32 +164,38 @@ describe("index", () => {
       expect(Sum.serialize(x)).toEqual(["B", null])
 
       fc.assert(
-        fc.property(fc.integer(), n =>
-          expect(f.decode(pipe(n, A, Sum.serialize))).toEqual(E.right(A(n))),
-        ),
+        fc.property(fc.integer(), n => {
+          return expect(f.decode(["A", n.toString()])).toEqual(E.right(A(n)))
+        }),
       )
     })
   })
 
   describe("getCodec", () => {
+    const nested = getCodecFromSerialized<Nested>()({
+      Nested: t.number,
+    })
     const f = getCodec<S>({
-      A: t.number,
+      // TODO: fix tests, or prevent this
+      A: NumberFromString,
```

Comment:
Not sure what to do about this.

Reply:
Don't we need `n` on line 190 to be converted to a string? In the test failure it says it's received `0` as a value which shouldn't decode per this codec.

Reply:
I think I tried that but I ran into more problems. 🤷‍♂️